forge-game/src/main/java/forge/game/ability/effects/AnimateAllEffect.java

```diff
@@ -133,15 +132,16 @@ public void resolve(final SpellAbility sa) {
 
         CardCollectionView list;
 
-        ZoneType z = sa.hasParam("Zone") ? ZoneType.smartValueOf(sa.getParam("Zone")) : ZoneType.Battlefield;
+        List<ZoneType> z = sa.hasParam("Zone") ? ZoneType.listValueOf(sa.getParam("Zone")) : 
+            ZoneType.listValueOf("Battlefield");
 
         if (sa.usesTargeting() || sa.hasParam("Defined")) {
             list = getTargetPlayers(sa).getCardsIn(z);
         } else {
             list = game.getCardsIn(z);
         }
 
-        list = CardLists.getValidCards(list, valid, sa.getActivatingPlayer(), host, sa);
+        list = AbilityUtils.filterListByType(list, valid, sa);
```

Comment:
I made this like PumpAllEffect – with this, Teferi's Contingency works. Is this OK?